src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py

```diff
@@ -285,22 +286,31 @@ def __init__(self, label_schema: LabelSchemaEntity):
         self.label_schema = label_schema
 
     def convert_to_annotation(
-        self, predictions: List[Tuple[int, float]], metadata: Optional[Dict] = None
+        self,
+        predictions: List[Tuple[int, float]],
+        metadata: Optional[Dict] = None,
+        hierarchical_info: Optional[Dict] = None,
     ) -> AnnotationSceneEntity:
         """Convert predictions to OTX Annotation Scene using the metadata.
 
         Args:
             predictions (tuple): Raw predictions from the model.
             metadata (Dict[str, Any]): Variable containing metadata information.
+            hierarchical_info (Dict): Info from model.hierarchical_info["cls_heads_info"]
 
         Returns:
             AnnotationSceneEntity: OTX annotation scene entity object.
         """
         labels = []
-        for index, score in predictions:
-            labels.append(ScoredLabel(self.labels[index], float(score)))
+
         if self.hierarchical:
+            hierarch_labels = get_hierarchical_label_list(hierarchical_info, self.labels)
```

Comment:
Can we move this to the converter's constructor?

Reply:
Good point, removed